### PR TITLE
Add colors@3.0.0 w/ npm auto-update & update package.json

### PR DESCRIPTION
**Pull request by samme**
Pull request for issue: none
Related issue(s): none

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/mrmrs/colors
 * Official website (optional, not the repository): http://clrs.cc
 * NPM package url (optional): https://www.npmjs.com/package/colors.css
 * License and its reference: MIT https://github.com/mrmrs/colors/blob/master/LICENSE.md
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 223
   - Count of stars: 7272
   - Count of forks: 439
 * NPM download stats (optional):
   - Downloads in the last day: ~500
   - Downloads in the last week: ~2800
   - Downloads in the last month: ~12900

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ ] This lib was not found on cdnjs repo
 * [X] No already exist / duplicated issue and PR
 * [X] The lib has notable popularity
   * [X] More than 100 [Stars, Watchers, & Forks] on [GitHub]
   * [X] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update) _(using npm auto-update instead)_
 * [X] Auto-update setup
 * [ ] Auto-update target/source is valid _(using npm auto-update instead)_.
 * [X] Auto-update filemap is correct.

# Git commit checklist
 * [X] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [X] The parent of the commit(s) in the PR is not old than 3 days.
 * [X] Pull request is sending from a non-master branch with meaningful name.
 * [X] Separate unrelated changes into different commits.
 * [X] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by clairetsai818** [comment]
Hi @samme!
Please minify the source files. 
Just use the commands below!
(1)  `git clone https://github.com/PeterDaveHello/web-minify-helper.git`
(2) `cd web-minify-helper`
(3) `npm install`
(4) `git submodule update --init`
(5) find the files you would like to minify
(6) `~/web-minify-helper/minify.sh`
Feel free to ask any question!
Thanks for your contribution!

**Comment by PeterDaveHello** [comment]
@clairetsai818 The 6th step, you assume the repository will be cloned at home directory, but you didn't mention it at all, and you don't even mention about what to do after you found the files need to be minified at 5th step.

**Comment by clairetsai818** [comment]
@PeterDaveHello I revised the step in a more clearly way.

File: ajax/libs/colors/package.json [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
+  },
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/mrmrs/colors.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "css/*",
```
Comment (hare1039) [inline ajax/libs/colors/package.json]: You can use a cooler way like
```
+(css|js)/*
```
You can try other rules if you want to.

**Comment by hare1039** [comment]
Also, would you make your commit message more detail?
like
```
Add colors@3.0.0 with git auto-update
```
```
Enrich and update package.json
```

We will appreciate if you separate commits of add `auto-update` with `package.json enrichment`
Thanks!

File: ajax/libs/colors/3.0.0/js/colors.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+(function() {
```
Comment (samme) [inline ajax/libs/colors/3.0.0/js/colors.js]: Yes, minified version is already included at https://github.com/samme/cdnjs/blob/95900c27e8b863edef7116b43244708cbe8217b2/ajax/libs/colors/3.0.0/js/colors.min.js

**Comment by samme** [comment]
Hi @clairetsai818, minified files are in

- https://github.com/samme/cdnjs/blob/95900c27e8b863edef7116b43244708cbe8217b2/ajax/libs/colors/3.0.0/css/colors.min.css
- https://github.com/samme/cdnjs/blob/95900c27e8b863edef7116b43244708cbe8217b2/ajax/libs/colors/3.0.0/js/colors.min.js

File: ajax/libs/colors/package.json (hunk 1 above)
Comment (samme) [inline ajax/libs/colors/package.json]: Can I leave it like this?

File: ajax/libs/colors/package.json [hunk 3]
```diff
@@ -1,23 +1,39 @@
 {
   "name": "colors",
-  "filename": "colors.min.css",
-  "version": "1.0",
-  "description": "Better colors for the web. A collection of skin classes for faster prototyping and nicer looking sites.",
+  "version": "3.0.0",
+  "filename": "js/colors.js",
```
Comment (clairetsai818) [inline ajax/libs/colors/package.json]: After you minify the file, lets's use `color.min.js` instead of `color.js` in the filename field.

File: ajax/libs/colors/package.json (hunk 1 above)
Comment (hare1039) [inline ajax/libs/colors/package.json]: I think its ok.
It is optional. :)

**Comment by PeterDaveHello** [comment]
@samme we didn't see you push the updates to the branch for this PR

**Comment by samme** [comment]
@clairetsai818 @hare1039 I pushed changes:

- `filename` now references `js/colors.min.js`
- split into 2 commits

**Comment by PeterDaveHello** [comment]
Hi @samme, looks like GitHub cannot recognize the author in the commit, would you like to confirm that the email is in your GitHub account? (Unless you don't really care about the contribution calculation) Thanks.

**Comment by samme** [comment]
Hi @PeterDaveHello I don't need the contribution count, thanks.

**Comment by PeterDaveHello** [comment]
@samme No problem, thanks for your reply!

File: ajax/libs/colors/package.json (hunk 3 above)
Comment (samme) [inline ajax/libs/colors/package.json]: Done.

**Comment by PeterDaveHello** [comment]
If the npm versions can be fetched via auto-update, yes.

File: ajax/libs/colors/package.json [hunk 4]
```diff
@@ -1,23 +1,36 @@
...
-  }
+  },
+  "license": "MIT",
+  "npmName": "colors.css",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "css/*",
+        "js/*"
```
Comment (extend1994) [inline ajax/libs/colors/package.json]: `css/*` and `js/*` works for current version, but I think `css/**/*` and `js/**/*` is a better choice. Because the former one can only fetch the files under `css` or `js`, can not fetch the file under sub folders. For example, there is a file path `css/images/img.jpg` then you can fetch this file using the former rule. However, it's possible that the author will add some files under `css` and `js` used in `colors.min.css` or `colors.min.js`.

**Comment by PeterDaveHello** [comment]
Hi @samme, thanks for your contribution, for the first commit, would you please mention the library name in the first line of the commit message? Thank you.

**Comment by PeterDaveHello** [comment]
Looks the same, something wrong?

File: ajax/libs/colors/package.json [hunk 5]
```diff
@@ -1,23 +1,36 @@
...
+    "postcss",
+    "palette",
+    "css",
+    "oocss"
   ],
   "repository": {
     "type": "git",
     "url": "https://github.com/mrmrs/colors.git"
   },
-  "license": "MIT",
```
Comment (maruilian11) [inline ajax/libs/colors/package.json]: hi @samme Could you please make the license unmodified?

Comment (samme) [inline ajax/libs/colors/package.json]: Hi @maruilian11 license field moved to [line 25](https://github.com/samme/cdnjs/blob/2aa19cd37abc624dee4739fbde7a541e5936497d/ajax/libs/colors/package.json#L25) but is unchanged.

Comment (PeterDaveHello) [inline ajax/libs/colors/package.json]: @samme we need to maintain the minimum diff, thanks.

**Comment by maruilian11** [comment]
@cdnjs/intern2 please help me make the double check.

File: ajax/libs/colors/package.json [hunk 6]
```diff
@@ -19,5 +22,15 @@
...
     "url": "http://mrmrs.cc"
-  }
+  },
+  "npmName": "colors.css",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "css/*",
+        "js/*"
```
Comment (pvnr0082t) [inline ajax/libs/colors/package.json]: It will be better to use `"+(js|css)/colors*"` instead of these two. Thanks.

Comment (samme) [inline ajax/libs/colors/package.json]: The results are identical.

Comment (pvnr0082t) [inline ajax/libs/colors/package.json]: For now, results are identical, but if there are unrelated files put into `css` and `js` directory in the future, we don't want to fetch them.

File: ajax/libs/colors/package.json [hunk 7]
```diff
@@ -19,5 +22,14 @@
     "name": "@mrmrs",
     "email": "hi@mrmrs.cc",
     "url": "http://mrmrs.cc"
-  }
+  },
+  "npmName": "colors.css",
```
Comment (pvnr0082t) [inline ajax/libs/colors/package.json]: @samme Adding npm auto-update part should in `Add colors@3.0.0 with npm auto-update` part, not in `Enrich and update colors package.json` part.

**Comment by PeterDaveHello** [comment]
Any updates here? Thanks!

**Comment by PeterDaveHello** [comment]
Hi @samme, would you like to finish it?

**Comment by samme** [comment]
I would like to, but I don't have time to rebase again right now.

**Comment by maruilian11** [comment]
Hi @samme, I am willing to send the PR for you if you don't have much time.
Just let me know if you need it. 😃

**Comment by samme** [comment]
@maruilian11 yes, that would be great, thanks.

**Comment by maruilian11** [comment]
@cdnjs/library-reviewer I got an Error when auto-update the lib from npm:
```
Looking for npm enabled libraries...
Found 1 npm enabled libraries
Checking versions for colors.css (colors)
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: invalid distance too far back
    at Zlib._handle.onerror (zlib.js:363:17)
```
Do you meet the same problem?

**Comment by sufuf3** [comment]
@maruilian11 I meet it before.
I have no time to figure it out why it happen. (maybe it is memory or swap something else let it happen.)
I reboot the machine and it's work again.

**Comment by maruilian11** [comment]
@sufuf3 Do you mean the problem solve after rebooting? Can you help try npm auto-update on this PR, too? I work on server, and it can't be rebooted. Thanks.

**Comment by PeterDaveHello** [comment]
I don't think it's related with rebooting.

**Comment by PeterDaveHello** [comment]
@sufuf3 please take this.

**Comment by sufuf3** [comment]
@cdnjs/library-reviewer I help modified the `package.json` with different commit messages.
Please help review this PR, Thank you.

**Comment by PeterDaveHello** [comment]
@sufuf3 ask review from @cdnjs/intern3 first.

**Comment by sufuf3** [comment]
@cdnjs/intern3 Please help review this PR, Thank you.

**Comment by PeterDaveHello** [comment]
@maruilian11 @pvnr0082t please help review this PR, thanks!

File: ajax/libs/colors/package.json [hunk 8]
```diff
@@ -1,14 +1,17 @@
 {
   "name": "colors",
-  "filename": "colors.min.css",
+  "filename": "js/colors.min.js",
   "version": "1.0",
-  "description": "Better colors for the web. A collection of skin classes for faster prototyping and nicer looking sites.",
+  "description": "Better default colors for the web. A collection of skin classes for faster prototyping and nicer looking sites.",
   "homepage": "http://clrs.cc",
   "keywords": [
-    "css",
```
Comment (pvnr0082t) [inline ajax/libs/colors/package.json]: @sufuf3 We don't need to do unnecessary changes. "css" should not be moved.

**Comment by sufuf3** [comment]
@pvnr0082t I modified the `css` part, and add v0.2.0, v2.0.0 from GitHub due to npm auto-update can't get them.
please help me review this PR, thanks!

**Comment by maruilian11** [comment]
hmm.. I still get problems when using the npm auto-update mechanism.
@sufuf3 So, you don't have this problem?

**Comment by sufuf3** [comment]
@maruilian11 I don't have this problem.

**Comment by sufuf3** [comment]
@cdnjs/intern3 @maruilian11 @pvnr0082t please help review this PR, thanks.

File: ajax/libs/colors/package.json [hunk 9]
```diff
@@ -1,14 +1,17 @@
 {
   "name": "colors",
-  "filename": "colors.min.css",
-  "version": "1.0",
-  "description": "Better colors for the web. A collection of skin classes for faster prototyping and nicer looking sites.",
+  "filename": "js/colors.min.js",
```
Comment (PeterDaveHello) [inline ajax/libs/colors/package.json]: This filename change should come with version 3.0.0, and mention that in the commit message. I don't want a single revision failed the test.

**Comment by sufuf3** [comment]
Done.

**Comment by PeterDaveHello** [comment]
@sufuf3 don't you need to request review from @cdnjs/library-reviewer ?

**Comment by sufuf3** [comment]
@cdnjs/intern3 @cdnjs/library-reviewer I modified the filename to the commit `	Add colors@3.0.0 w/ npm auto-update` as this comment(https://github.com/cdnjs/cdnjs/pull/10727#discussion_r125230786) said
Please help me review this PR, Thank you.

**Comment by maruilian11** [comment]
hmm... It seems that I can't re-approve. 
I think this PR is ok.
@sufuf3 Please rebase the branch with our latest master if you request any review next time. Thanks.

**Comment by PeterDaveHello** [comment]
@maruilian11 you should re-request review from yourself first.

**Comment by sufuf3** [comment]
@extend1994 I add "Add" 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@PeterDaveHello I updated the commit message.
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@maruilian11 as the comment I just met https://github.com/cdnjs/cdnjs/pull/10727#issuecomment-304043698
I found that if I remove the folder <library_name> which under `/run/shm/cdnjs_NPM_temp/` and rerun 
`./auto-update.js run <library_name>` again. It's work!
I met the error 'A' as the followings which as same with you after I met the 'B' 
- 'A' (second error message)
```
Looking for npm enabled libraries...
Found 1 npm enabled libraries
Checking versions for colors.css (colors)
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: invalid distance too far back
    at Zlib._handle.onerror (zlib.js:363:17)
```
- 'B' (first error message)
```
Looking for npm enabled libraries...
Found 1 npm enabled libraries
Checking versions for colors.css (colors)
events.js:141
      throw er; // Unhandled 'error' event
      ^
Error: ENOENT: no such file or directory, open '/run/shm/cdnjs_NPM_temp/<library_name>/<version>/dist.tar.gz'
    at Error (native)
```
Although it's a little bit wired that I can find the `dist.tar.gz` which under the error message said `/run/shm/cdnjs_NPM_temp/<library_name>/<version>/`.

**Comment by maruilian11** [comment]
@ thanks, I will try this way next time I meet the problem.

**Comment by maruilian11** [comment]
And rebase this branch with our latest master branch.

**Comment by PeterDaveHello** [comment]
@extend1994 can you take a look?

**Comment by PeterDaveHello** [comment]
So colors `v1.0` has different structure with all the others even `v0.2.0`? Can you also fix it here? Don't delete the old one but just add the correct one.

@maruilian11 @extend1994 @sufuf3 : 
```
0.2.0
├── css
│   ├── colors.css
│   └── colors.min.css
└── js
    ├── colors.js
    └── colors.min.js
1.0
└── colors.min.css
2.0.0
├── css
│   ├── colors.css
│   └── colors.min.css
└── js
    ├── colors.js
    └── colors.min.js
3.0.0
├── css
│   ├── colors.css
│   └── colors.min.css
└── js
    ├── colors.js
    └── colors.min.js
```

**Comment by sufuf3** [comment]
@cdnjs/intern3 @cdnjs/library-reviewer 
Please help me review this PR, Thank you.

Memo:
I can't find the `v1.0` in this library release versions. I found `v1.0.0` from npm. but the `colors.min.css` of `v1.0` md5sum is different from `colors.min.css` of `v1.0.0`.
So I copied `1.0/colors.min.css` to ``1.0/css/colors.min.css`.
And let auto-update add `1.0.0` folder.

I also check the file history, there is nothing about the file checksum. Just know that it added from GitHub repo. but I can't find `v1.0` git tag from GitHub repo. (#2427)

I didn't remove the `1.0/colors.min.css` as peter's comment. Also, I think maybe someone use this file via our CDNJS, so I didn't remove the `1.0/colors.min.css`.

**Comment by maruilian11** [comment]
```
Manually add v1.0 from colors/1.0/colors.min.css to
colors/1.0/css/colors.min.css due to different file structure
```

---> 

```
Copy 1.0/colors.min.css as 1.0/css/colors.min.css to satisfy current file structure
```
will be better

**Comment by PeterDaveHello** [comment]
@sufuf3 your commit message looks partially duplicated with themselves? Can we avoid that?

**Comment by sufuf3** [comment]
Sorry, I'll be careful next time.
I updated the commit message
And I use `Update v1.0 of colors` because v1.0 is already host, I just copy `1.0/colors.min.css` as `1.0/css/colors.min.css` to fit current file structure.

@cdnjs/intern3 @cdnjs/library-reviewer 
Please help me review this PR, Thank you.

**Comment by PeterDaveHello** [comment]
Still in https://github.com/cdnjs/cdnjs/pull/10727/commits/16d86bdd6a9be093af7eade5d38c84e95ce39871, and can we just use `Update description, keywords info of "colors" library` for https://github.com/cdnjs/cdnjs/pull/10727/commits/84819ce8572f0190f366010d519d43ca2136686b? Isn't it a commit message can fit one line? Then why should be use 2 lines?

**Comment by sufuf3** [comment]
Updated the commit message
@cdnjs/intern3 
Please help me review this PR, Thank you.
